BrainPortal/app/helpers/userfiles_helper.rb

```diff
@@ -77,16 +77,24 @@ def file_link_table(previous_userfile, next_userfile, sort_index, options = {})
   end
 
   # Generates links to pretty file content for userfiles
-  # of type TextFile or ImageFile; this method is going to
-  # be replaced by a proper generic framework in 4.3.0 !
+  # of type TextFile or ImageFile
+  # Generates download link for any other type of file
   def data_link(file_name, userfile)
-    display_name  = Pathname.new(file_name).basename.to_s
+    dp = DataProvider.find(userfile.data_provider_id)
```

Comment:
It's unnecessary to find the DP and call the cache_full_path method on it, the userfile model already has shortcuts:

```ruby
full_path = userfile.cache_full_path.dirname + file_name
```

(because cache_full_path already returns a Pathname object)